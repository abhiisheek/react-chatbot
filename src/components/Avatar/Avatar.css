@import '../_colors.css';
@import '../_variables.css';

.icon {
  border: 1px solid var(--border-color);
  border-radius: 12px;
  height: var(--avatar-icon-height);
  width: var(--avatar-icon-width);
  display: flex;
  align-items: center;
  justify-content: center;
}
