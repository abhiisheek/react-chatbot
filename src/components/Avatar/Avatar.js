import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import styles from './Avatar.css';

/** A component to show the avatar */
class Avatar extends PureComponent {
  render() {
    return <div className={`${this.props.icon} ${styles.icon}`} />;
  }
}

Avatar.displayName = 'Avatar';

Avatar.propTypes = {
  /** Classname of the avatar/icon to be shown */
  icon: PropTypes.string.isRequired
};

export default Avatar;
