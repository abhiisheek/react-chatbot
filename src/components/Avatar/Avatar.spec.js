import React from 'react';
import renderer from 'react-test-renderer';

import { shallow } from 'enzyme';

import Avatar from './Avatar';

describe('Avatar', () => {
  it('Should have displayName set on the component', () => {
    expect(Avatar.displayName).toBeDefined();
  });

  it('Snapshot test', () => {
    const snapshot = renderer.create(<Avatar icon={'test'} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('Should load with person-icon', () => {
    const wrapper = shallow(<Avatar icon="person-icon" />);
    expect(wrapper.find('.person-icon').length).toBe(1);
  });
});
