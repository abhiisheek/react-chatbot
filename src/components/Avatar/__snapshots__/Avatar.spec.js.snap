// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Snapshot test 1`] = `
<div
  className="test icon"
/>
`;
